modules/dnn/include/opencv2/dnn/dnn.hpp

```diff
@@ -996,6 +996,45 @@ CV__DNN_EXPERIMENTAL_NS_BEGIN
                              CV_OUT std::vector<int>& indices,
                              const float eta = 1.f, const int top_k = 0);
 
+     class CV_EXPORTS_W Model : public Net
+     {
+     public:
+         Model(const std::string& model, const std::string& config = "");
+         Model(const Net& network);
+
+         Model& setInputSize(const Size& size);
+         Model& setInputSize(int width, int height);
+         Model& setInputMean(const Scalar& mean);
+         Model& setInputScale(float scale);
+         Model& setInputCrop(bool crop);
+         Model& setInputSwapRB(bool swapRB);
+         void predict(InputArray frame, OutputArrayOfArrays outs);
+
+     protected:
+         struct Impl;
+         Ptr<Impl> impl_;
```

Comment:
Maybe it's better to name it as `impl`? Like for dnn::Net

Reply:
done